src/View/SSTemplateEngine.php

```diff
@@ -0,0 +1,468 @@
+     * Used by the <% include %> template tag to process included templates.
+     *
+     * @param array $overlay Associative array of fields (e.g. args into an include template) to inject into the
+     * template as properties. These override properties and methods with the same name from $data and from global
+     * template providers.
+     */
+    public static function execute_template(array|string $template, ViewLayerData $data, array $overlay = [], ?SSViewer_Scope $scope = null): string
+    {
+        $engine = static::create($template);
+        return $engine->render($data, $overlay, $scope);
+    }
+
+    /**
+     * Triggered early in the request when someone requests a flush.
+     */
+    public static function flush(): void
+    {
+        SSTemplateEngine::flushTemplateCache(true);
+        SSTemplateEngine::flushCacheBlockCache(true);
+    }
+
+    /**
+     * Clears all parsed template files in the cache folder.
+     *
+     * @param bool $force Set this to true to force a re-flush. If left to false, flushing
+     * will only be performed once a request.
+     */
+    public static function flushTemplateCache(bool $force = false): void
+    {
+        if (!SSTemplateEngine::$template_cache_flushed || $force) {
+            $dir = dir(TEMP_PATH);
+            while (false !== ($file = $dir->read())) {
+                if (strstr($file ?? '', '.cache')) {
+                    unlink(TEMP_PATH . DIRECTORY_SEPARATOR . $file);
+                }
+            }
+            SSTemplateEngine::$template_cache_flushed = true;
+        }
+    }
+
+    /**
+     * Clears all partial cache blocks.
+     *
+     * @param bool $force Set this to true to force a re-flush. If left to false, flushing
+     * will only be performed once a request.
+     */
+    public static function flushCacheBlockCache(bool $force = false): void
+    {
+        if (!SSTemplateEngine::$cacheblock_cache_flushed || $force) {
+            $cache = Injector::inst()->get(CacheInterface::class . '.cacheblock');
+            $cache->clear();
+            SSTemplateEngine::$cacheblock_cache_flushed = true;
+        }
+    }
+
+    public function hasTemplate(array|string $templateCandidates): bool
+    {
+        return (bool) $this->findTemplate($templateCandidates);
+    }
+
+    public function renderString(string $template, ViewLayerData $model, array $overlay = [], bool $cache = true): string
+    {
+        $hash = sha1($template);
+        $cacheFile = TEMP_PATH . DIRECTORY_SEPARATOR . ".cache.$hash";
+
+        // Generate a file whether we're caching or not.
+        // This is an inefficiency that's required due to the way rendered templates get processed.
+        if (!file_exists($cacheFile) || Injector::inst()->get(Kernel::class)->isFlushed()) {
+            $content = $this->parseTemplateContent($template, "string sha1=$hash");
+            $fh = fopen($cacheFile, 'w');
+            fwrite($fh, $content);
+            fclose($fh);
+        }
```

Comment:
A follow-up card will likely include changing how template caching works... I hate that this is creating a file that it might not even need, but refactoring that is not in scope for this PR.